internal/database/mysql.go

```diff
@@ -120,59 +118,73 @@ func (db *MySQLDatabase) GetUser(id int) (user *internal.User) {
+func (db *MySQLDatabase) getOpenID() (id uint64, err error) {
+	err = db.underlyingDB.QueryRow("SELECT id FROM openid LIMIT 1").Scan(&id)
 	if err != nil && !errors.Is(err, sql.ErrNoRows) {
 		log.Println("Failed to query for an openid -", err)
+		return 0, errors.New("invalid openid")
 	}
-	return
+	return id, nil
 }
 
-func (db *MySQLDatabase) AddUser(username string, email string, password string) (err error) {
-
-	// TODO: There needs to be a check if the usernames/emails/passwords empty here.
-	if len(username) == 0 || len(email) == 0 || len(password) == 0 {
-		return errors.New("empty fields")
-	}
-	// TODO(@seoyoungcho213): For efficiency, we might be able to return the new id here with only a single query?``
-	query := "INSERT INTO users (username, email, password"
-	if openID := db.getOpenID(); openID == 0 {
-		log.Println("All existing IDs in use, assigning new ID to {" + username + "}")
-		query += `) VALUES ("%s", "%s", "%s")`
+func (db *MySQLDatabase) AddUser(first_name, middle_name, last_name, email, password, date_of_birth string) (
+	id uint64, err error) {
+	if len(first_name) == 0 || len(last_name) == 0 || len(email) == 0 || len(password) == 0 || len(date_of_birth) == 0 {
+		return 0, errors.New("empty fields")
+	}
+	query := "INSERT INTO users (first_name, middle_name, last_name, email, password, date_of_birth"
+	if openID, _ := db.getOpenID(); openID == 0 {
+		log.Println("All existing IDs in use, assigning new ID to {" + email + "}")
+		query += `) VALUES (?, ?, ?, ?, ?, ?)`
 	} else {
-		log.Println("Re-using open id:", openID, "for {"+username+"}")
-		query += `, id) VALUES ("%s", "%s", "%s", ` + strconv.Itoa(openID) + `)`
+		log.Println("Re-using open id:", openID, "for {"+email+"}")
+		id = openID
+		query += `, id) VALUES (?, ?, ?, ?, ?, ?, ` + strconv.FormatUint(id, 10) + `)`
+		if _, err = db.underlyingDB.Exec("DELETE FROM openid WHERE id = ?", id); err != nil {
+			log.Println("Error deleting open id -", err)
+			return id, errors.New("internal server error")
+		}
 	}
 	hashedPassword, err := bcrypt.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
 	if err != nil {
 		log.Println("Error hashing password -", err)
-		return errors.New("internal server error")
+		return id, errors.New("internal server error")
 	}
-	if _, err = db.underlyingDB.Exec(fmt.Sprintf(query, username, email, hashedPassword)); err != nil {
+	result, err := db.underlyingDB.Exec(query, first_name, middle_name, last_name, email, hashedPassword, date_of_birth)
+	if err != nil {
 		log.Println("Error adding user -", err)
-		return errors.New("internal server error")
+		return 0, errors.New("internal server error")
 	}
-	// TODO(@seoyoungcho213): Make sure to delete the openID if used here ("delete if exists" would be perfect).
-	return
+	if id == 0 {
+		userid, err := result.LastInsertId()
+		if err != nil {
+			log.Println("Error getting user ID -", err)
+			return 0, errors.New("internal server error")
+		}
+		id = uint64(userid)
```

Comment:
Move the log from line 137 here, and log the id as part of the message, make sure to follow the log.println formats